ddtrace/contrib/wsgi/wsgi.py

```diff
@@ -115,45 +117,44 @@ def __call__(self, environ, start_response):
-            app_span.set_tag_str("component", self._config.integration_name)
-
-            intercept_start_response = functools.partial(
-                self._traced_start_response, start_response, req_span, app_span
+            req_span = self.tracer.trace(
+                self._request_span_name,
+                service=trace_utils.int_service(self._pin, self._config),
+                span_type=SpanTypes.WEB,
             )
-            result = self.app(environ, intercept_start_response)
-            self._application_span_modifier(app_span, environ, result)
-            app_span.finish()
-        except BaseException:
-            req_span.set_exc_info(*sys.exc_info())
-            app_span.set_exc_info(*sys.exc_info())
-            app_span.finish()
-            req_span.finish()
-            raise
-        # start flask.response span. This span will be finished after iter(result) is closed.
-        # start_span(child_of=...) is used to ensure correct parenting.
-        resp_span = self.tracer.start_span(self._response_span_name, child_of=req_span, activate=True)
 
-        # set component tag equal to name of integration
-        resp_span.set_tag_str("component", self._config.integration_name)
+            # set component tag equal to name of integration
+            req_span.set_tag_str("component", self._config.integration_name)
+
+            self._request_span_modifier(req_span, environ)
+            try:
+                app_span = self.tracer.trace(self._application_span_name)
+                # set component tag equal to name of integration
+                app_span.set_tag_str("component", self._config.integration_name)
+                intercept_start_response = functools.partial(
+                    self._traced_start_response, start_response, req_span, app_span
+                )
+                result = self.app(environ, intercept_start_response)
+                self._application_span_modifier(app_span, environ, result)
+                app_span.finish()
+            except BaseException:
+                req_span.set_exc_info(*sys.exc_info())
+                app_span.set_exc_info(*sys.exc_info())
+                app_span.finish()
+                req_span.finish()
+                raise
+            # start flask.response span. This span will be finished after iter(result) is closed.
+            # start_span(child_of=...) is used to ensure correct parenting.
+            resp_span = self.tracer.start_span(self._response_span_name, child_of=req_span, activate=True)
 
-        self._response_span_modifier(resp_span, result)
+            # set component tag equal to name of integration
+            resp_span.set_tag_str("component", self._config.integration_name)
+            self._response_span_modifier(resp_span, result)
 
-        return _TracedIterable(iter(result), resp_span, req_span)
+            return _TracedIterable(iter(result), resp_span, req_span)
+        finally:
+            _asm_context.reset()
```

Comment:
are we able to push enough context from the wsgi environ to make a blocking decision?

if someone wrapped their app with the wsgi middleware then they could prevent the request from even calling their web app in the first place.

("nice to have"? feel free to resolve with no action)

Reply:
We get the information on `flask_patch::request_tracer` (called before this) since that's the point where we have the request information.